Remove legacy InstantSend code

codablock opened the pull request:
This PR removes all legacy InstantSend code, including UI and RPC stuff.
See individual commits for details.

UdjinM6 commented on src/qt/res/css/crownium.css:
```diff
@@ -1290,17 +1290,6 @@ padding-top:20px;
-padding-top:20px;
-padding-bottom:18px;
-margin-left:10px;
-margin-right:20px;
```
`margin-right` should now be applied to `QDialog#SendCoinsDialog .QCheckBox#checkUsePrivateSend` to keep PS checkbox away from "Balance:" label, see https://github.com/UdjinM6/dash/commit/b9d834da52f87464538edaef8c28395dc414ed88.

codablock replied:
Thanks, added it :)

UdjinM6 commented on src/qt/transactiondesc.cpp:
```diff
@@ -57,25 +55,9 @@ QString TransactionDesc::FormatTxStatus(const CWalletTx& wtx)
-        } else {
-            strTxStatus += tr("InstantSend verification failed");
+        if (wtx.IsLockedByInstantSend()) {
+            strTxStatus += " (" + tr("verified via InstantSend") + ")";
```
Is this intentional? Won't this cause confusion for people who are used to see `LLMQ based` here? (also, should we remove `LLMQ based` for ChainLocks too in some future PR to keep things consistent?)

codablock replied:
It was intentional, but yeah, I guess you're right that this could cause confusion. I'll revert that part.

PastaPastaPasta replied:
I'm not sure tacking on LLMQ to these features is good from a usability perspective. LLMQ looks crazy, InstantSend anyone can understand.

codablock replied:
At least for the next release, community members might get confused when they see it changing back to just "InstantSend". I communicated a few times that we might decide to disable LLMQ based InstantSend if it turns out to not be good enough, so this is how such a change could be interpreted. But it should be fine to change it in the release following the next one.

codablock commented on src/protocol.cpp:
```diff
@@ -243,8 +243,8 @@ const char* CInv::GetCommandInternal() const
         case MSG_FILTERED_BLOCK:            return NetMsgType::MERKLEBLOCK;
-        case MSG_TXLOCK_REQUEST:            return NetMsgType::TXLOCKREQUEST;
         case MSG_TXLOCK_VOTE:               return NetMsgType::TXLOCKVOTE;
+        case MSG_LEGACY_TXLOCK_REQUEST:     return NetMsgType::TXLEGACYLOCKREQUEST;
```
Agree and renamed it to TXLEGACYLOCKREQUEST

codablock commented:
@PastaPastaPasta Nope, spork 5, 6 and 20 are removed instead. Spork 20 was only meant to switch to the new system, but spork 2 is still meant to signal if InstantSend is enabled or not. But...I just realized that we have to clarify with the mobile team if they actually handle sporks this way...

codablock commented:
Just clarified with the mobile team that spork20 can't be easily removed as they internally still depend on it. We can remove spork20 in a few months however. I've reverted removal of spork20 and squashed that into the original commit.

UdjinM6 commented:
Should also fix tests now

codablock commented:
Not sure I understand...you assume that tests will fail?

UdjinM6 commented:
Oh, you only reverted spork20 _existence_, not the _logic_ behind it, I see.

codablock commented:
Exactly, this is really just to ensure that the spork is propagated to clients/nodes that still need it.

UdjinM6 commented:
needs rebase

PastaPastaPasta commented on src/qt/walletmodel.cpp:
```diff
@@ -414,12 +390,7 @@ WalletModel::SendCoinsReturn WalletModel::sendCoins(WalletModelTransaction &tran
-            strCommand = NetMsgType::TXLOCKREQUEST;
-        }
-        if(!wallet->CommitTransaction(*newTx, *keyChange, g_connman.get(), state, strCommand))
+        if(!wallet->CommitTransaction(*newTx, *keyChange, g_connman.get(), state))
```
`if(` -> `if (`

codablock replied:
Out of scope of this PR...wallet.cpp is generally loaded with style infringements, but fixing them would lead to many merge conflicts in backports.

codablock replied:
Ah sorry, this is walletmodel.cpp and not wallet.cpp

PastaPastaPasta commented on src/wallet/rpcwallet.cpp:
```diff
@@ -67,15 +66,14 @@ void WalletTxToJSON(const CWalletTx& wtx, UniValue& entry)
-    entry.push_back(Pair("instantlock_internal", fLocked || fLLMQLocked));
+    entry.push_back(Pair("instantlock", fLocked || chainlock));
+    entry.push_back(Pair("instantlock_internal", fLocked));
     entry.push_back(Pair("chainlock", chainlock));
```
I believe these should use emplace_back instead of push_back (out of scope for this PR)

codablock replied:
Yepp, out of scope

PastaPastaPasta commented on src/protocol.h:
```diff
@@ -355,8 +354,8 @@ enum GetDataMsg {
     // NOTE: declare non-implmented here, we must keep this enum consistent and backwards compatible
-    MSG_TXLOCK_VOTE = 5,
     MSG_LEGACY_TXLOCK_REQUEST = 4,
+    /* MSG_TXLOCK_VOTE = 5, Legacy InstantSend and not used anymore  */
```
Why not just remove instead of commenting it out to forget about later? :laughing:

codablock replied:
Hmm...I tend to prefer keeping such comments when there is a chance that a message with these types appears. Makes it easier to look up what's going on.

PastaPastaPasta replied:
:+1:

codablock commented on src/protocol.cpp:
```diff
@@ -263,45 +226,59 @@ CInv::CInv(int typeIn, const uint256& hashIn)
+        case MSG_QUORUM_CONTRIB:            return NetMsgType::QCONTRIB;
+        case MSG_QUORUM_COMPLAINT:          return NetMsgType::QCOMPLAINT;
+        case MSG_QUORUM_JUSTIFICATION:      return NetMsgType::QJUSTIFICATION;
+        case MSG_QUORUM_PREMATURE_COMMITMENT: return NetMsgType::QPCOMMITMENT;
```
Yeah I know, but I thought that in this case it wouldn't be worth it to move everything to the right just because of this one line

PastaPastaPasta replied:
Imo if we want almost all of them in line then all of them should be in line.

codablock replied:
Hmm, ok agreed :) Changed this and squashed into original commits.

PastaPastaPasta replied:
:+1:

codablock commented on src/protocol.cpp:
```diff
@@ -263,45 +226,59 @@ CInv::CInv(int typeIn, const uint256& hashIn)
 {
-    if (!IsKnownType())
+    auto cmd = GetCommandInternal();
+    if (!cmd) {
```
Hmm...what part do you find hard to read and what would you prefer?

PastaPastaPasta replied:
Likely change to `

codablock replied:
But `cmd` is not a `std::string` but a `const char*`

PastaPastaPasta replied:
At least `pointer==nullptr` instead of `!pointer`

codablock replied:
I'll change it in this instance, but generally this is something that is sprinkled all over the source already, simply because it's quite typical for C++ code, independent from it being bad style or not.

codablock commented:
Rebased and force-pushed.

codablock commented:
FYI, this should help re-reviewing after the force-push:
```bash
$ function gfd() {
        local fp1=$(git merge-base --fork-point develop $1)
        local fp2=$(git merge-base --fork-point develop $2)
        echo fp1=$fp1
        echo fp2=$fp2
        diff --color=always -u -I'^[^-+]' <(git diff $fp1..$1) <(git diff $fp2..$2)
}
$ gfd b18ff68fc6c9289a8e08db29cad9228904c3e350 38d1aeabcc3133c00971f2ee0a95a2b4e6e52da2
```

This will show a diff of the diffs and thus filters out all the noise that you usually have when looking at a diff like in the "force-pushed" links created by Github. It however requires you to have the old commit locally, e.g. because you pulled the PR or my branch locally.

PastaPastaPasta commented on src/qt/transactiondesc.cpp:
```diff
@@ -57,25 +55,9 @@ QString TransactionDesc::FormatTxStatus(const CWalletTx& wtx)
 
-        if (wtx.IsLockedByLLMQInstantSend()) {
+        if (wtx.IsLockedByInstantSend()) {
             strTxStatus += " (" + tr("verified via LLMQ based InstantSend") + ")";
```
I'm not sure tacking on LLMQ to these features is good from a usability perspective. LLMQ looks crazy, InstantSend anyone can understand.
( This had another comment thread but was marked resolved and outdated so bringing it up here )

codablock replied:
I answered in the original thread

PastaPastaPasta replied:
Your initial idea was to remove the "via LLMQ" udjin thought we should keep it so you reverted that change and I'm saying we should get rid of it as you originally suggested since "LLMQ" isn't consumer friendly

codablock replied:
Did you read my answer in the original thread? (https://github.com/dashpay/dash/pull/3020#discussion_r301221967)

I generally agree that we should get rid of it, but I also agree to what udjin says as it might confuse people when we do it in the next release.

UdjinM6 replied:
I agree as well btw :) Let's keep this out of the scope of this specific PR and remove "LLMQ" part from both IS and ChainLocks strings in some another PR in the future.

PastaPastaPasta replied:
Okay, agreed for later

PastaPastaPasta replied:
Sorry about that @codablock, for some reason either I missed your comment or github wasn't showing it to me, because I looked for a response just before I made this comment thread

codablock replied:
Don't worry, happened to me as well already. Github seems to get confused with comments/threads when force-pushes are involved. Best is to enable email notifications for comments/mentions, this way you never miss a comment that is happening in one of the threads (and PRs) where you're involved.

UdjinM6 commented:
Needs rebase again.

codablock commented:
Rebased and force-pushed. Also re-added SPORK_16_INSTANTSEND_AUTOLOCKS as it is checked by a few mobile clients to decide on necessary fees.
